traits/traits_listener.py

```diff
@@ -990,20 +981,50 @@ def parse(self):
                     handler=self.handler,
                     wrapped_handler_ref=self.wrapped_handler_ref,
                     dispatch=self.dispatch,
+                    priority=self.priority,
+                    # Bug-for-bug compatibility with old behaviour: don't
+                    # propagate the 'deferred' or 'handler_type' values for the
+                    # child item.
```

Comment:
Should we link https://github.com/enthought/traits/issues/537 here? That looks like the relevant issue

Reply:
Will do. And yes, that's definitely the relevant issue for the `type`; not sure whether the `deferred` behaviour is actually correct or not.

Reply:
Done.